README.md

```diff
@@ -1,9 +1,87 @@
-### Display Locking
+# Display Locking
 
-This repo is a collection of documents, sample code, and discussions surrounding
-the Display Locking proposal.
+This is a short proposal for Display Locking. For more details, see the
+[explainer](https://github.com/chrishtr/display-locking/blob/master/explainer.md)
 
-- [Display Locking
-explainer](https://github.com/chrishtr/display-locking/blob/master/explainer.md)
+### Problem
+
+Websites are frequently relying on dynamic content to present information. This
+means that DOM is often manipulated using script to present rich and dynamic
+content. There are cases when this cause cause slow rendering phase updates,
+including style updates, layout, and paint. This, in turn, causes jank or
+noticeable delay when presenting content, because rendering phase is updated
+synchronously with user interactions and requestAnimationFrame script.
+
+The following are a few of the common patterns that can cause jank:
+- Resizing multi-pane UI with complex layout within each pane (e.g. IDEs)
+- Adding widgets with complex DOM.
+- Dynamically displaying content based on the current value of text input.
+- Measuring layout of otherwise hidden DOM with intent of sizing containers.
+
+Developers are aware of these issues and build systems to avoid adding complex
+DOM at once in order to prevent jank. As an example, ReactJS is [adding
+ability](https://reactjs.org/blog/2018/03/01/sneak-peek-beyond-react-16.html)
+to do async operations to help with this problem.
+
+### Proposal
+
+This document proposes to augment user-agent APIs to help address the problem of
+slow rendering phase updates. Specifically, we propose a concept of display
+locking an element.
+
+If an element is locked for display, it means that any DOM
+updates to its subtree are processed co-operatively, yielding periodically to
+allow script or user interactions to happen. 
+
+The visual content of an element which is locked for display is fixed to
+whatever content was present at the time the lock was acquired, and remains this
+way until the lock is released and all associated promises are resolved.
+
+### Example
+
+```js
+async function updateDom() {
+  let element = document.getElementById("container");
+
+  // Get and acquire a lock, causing the element to be locked.
+  let lock = element.getDisplayLock();
+  await lock.acquire();
+
+  // Modify element's subtree, which is processed co-operatively.
```

Comment:
This is not cooperative. DOM is updated right away, it's the rendering that is not updated until commit.